### PR TITLE
Executing LocalLimitExec with no column should not return an Err

### PR DESCRIPTION
**Pull request by kazuyukitanimura**
# Which issue does this PR close?
Closes https://github.com/apache/arrow-datafusion/issues/5701

# Rationale for this change
Bug fix

# What changes are included in this PR?
Use `options` to explicitly set `row_count`, closely related / similar solution to https://github.com/apache/arrow-datafusion/pull/4912

# Are these changes tested?
Yes

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/src/physical_plan/limit.rs [hunk 1]
```diff
@@ -462,7 +462,8 @@ impl LimitStream {
                 .iter()
                 .map(|col| col.slice(0, col.len().min(batch_rows)))
                 .collect();
-            Some(RecordBatch::try_new(batch.schema(), limited_columns).unwrap())
+            let options = RecordBatchOptions::new().with_row_count(Option::from(batch_rows));
```
Comment (comphead) [inline datafusion/core/src/physical_plan/limit.rs:465]: Thanks @kazuyukitanimura 
extending call `RecordBatch` to be called with options gives more flexibility

Comment (alamb) [reply datafusion/core/src/physical_plan/limit.rs:465]: I agree it is nice

**Comment by kazuyukitanimura** [comment]
cc @viirya @sunchao

**Review by alamb — COMMENTED** [review]
Thank you for the contribution @kazuyukitanimura (and the assist @comphead )

Would it be possible to make an end to end reproducer (either with SQL or with the DataFrame API) to show the problem? It would be nice to  ensure there aren't other issues with creating zero column batches.

I tried to reproduce the error and I was not able to:

```
DataFusion CLI v20.0.0
❯ create table foo(x int);
0 rows in set. Query took 0.004 seconds.
❯ insert into foo values(1);
0 rows in set. Query took 0.003 seconds.
❯ insert into foo values(2);
0 rows in set. Query took 0.002 seconds.
❯ explain select 1 from foo limit 1;
+---------------+---------------------------------------------------+
| plan_type     | plan                                              |
+---------------+---------------------------------------------------+
| logical_plan  | Projection: Int64(1)                              |
|               |   Limit: skip=0, fetch=1                          |
|               |     TableScan: foo projection=[x], fetch=1        |
| physical_plan | ProjectionExec: expr=[1 as Int64(1)]              |
|               |   GlobalLimitExec: skip=0, fetch=1                |
|               |     MemoryExec: partitions=1, partition_sizes=[2] |
|               |                                                   |
+---------------+---------------------------------------------------+
2 rows in set. Query took 0.004 seconds.
❯ select 1 from foo limit 1;
+----------+
| Int64(1) |
+----------+
| 1        |
+----------+
```

**Comment by kazuyukitanimura** [comment]
> Thank you for the contribution @kazuyukitanimura (and the assist @comphead )
> 
> Would it be possible to make an end to end reproducer (either with SQL or with the DataFrame API) to show the problem? It would be nice to ensure there aren't other issues with creating zero column batches.

Thanks @alamb that will have to be a separate work from this PR. I have some examples, but I need to think through how to remove some proprietary code from that. Additionally I suspect there will be other execs that have the same issue, so I would like make the reproducer generic. Would you mind approving this PR first given this is pretty much the same as https://github.com/apache/arrow-datafusion/pull/4912
I will file a ticket for the end. to end reproducer.

**Review by alamb — APPROVED** [review]
I think the real value in an end to end reproducer is so that we don't accidentally break the usecase in some other later refactor

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
To have end to end reproducer, you need to be able to project empty relation, not sure if DataFusion allows such thing in SQL/DataFrame API.

**Review by sunchao — APPROVED** [review]

**Comment by kazuyukitanimura** [comment]
Can anyone help with merging this unless there are more feedbacks?

**Pull request merged**